File: c/textgame.c
#include <stdio.h>
#include <stdlib.h>

void game();
void game_output(char map[], char message[], int coins, int pos);
void menu();

void game() {
    /* init vars */
    int coins       =   0;
    int pos         =   4;
    char map[9]     =   {'_','_','_','_','_','_','_','_','_'};
    char *message   =   "";
    char move;
    char *help_text =   "Moves:\n"
                        "e - move east\n"
                        "h - show help text\n"
                        "l - look around the room\n"
                        "n - move north\n"
                        "q - quit game\n"
                        "s - move south\n"
                        "t - touch something\n"
                        "w - move west\n";

    /* main game loop */
    for (;;) {
        /* set player location */
        map[pos] = '*';
        
        /* print game output and prompt */
        game_output(map, message, coins, pos);
        scanf(" %c", &move);
        system("clear");
        
        /* reset player location and message */
        map[pos]    = '_';
        message     = "";
        
        /* player moves */
        if (move == 'n' && pos > 2)
            pos -= 3;
        else if (move == 's' && pos < 6)
            pos += 3;
        else if (move == 'e' && pos != 2 && pos != 5 && pos != 8)
                pos += 1;
        else if (move == 'w' && pos != 0 && pos != 3 && pos != 6)
            pos -= 1;
        else if (move == 'h')
            message = help_text;
        else if (move == 'q')
            exit(0);
    }
}

void game_output(char map[], char message[], int coins, int pos) {
    /* print values and map */
    printf("Coins: %d\n", coins);
    printf(" _____ \n|%c|%c|%c|\n|%c|%c|%c|\n|%c|%c|%c|\n", \
    map[0],map[1],map[2],map[3],map[4],map[5],map[6],map[7],map[8]);
    
    /* print message and room text */
    printf("%s", message);
    if (pos == 0)
        printf("Room 0");
    if (pos == 1)
        printf("Room 1");
    if (pos == 2)
        printf("Room 2");
    if (pos == 3)
        printf("Room 3");
    if (pos == 4)
        printf("Room 4");
    if (pos == 5)
        printf("Room 5");
    if (pos == 6)
        printf("Room 6");
    if (pos == 7)
        printf("Room 7");
    if (pos == 8)
        printf("Room 8");
    printf("\n> ");
}

void menu() {
    /* menu vars */
    int choice;
    char *about_text =  "You awake in a strange building with nine rooms,\n"
                        "connected together by many doors. Each room has one\n"
                        "coin hidden within. Find all the coins to escape!\n";

    /* menu text */
    printf("9coins - By: Vesche\n");
    printf("1. Play\n2. About\n3. Quit\n");

    /* menu loop */
    for (;;) {
        /* menu prompt */
        printf("> ");
        scanf("%d", &choice);
        
        /* menu choices */
        if (choice == 1)
            break;
        else if (choice == 2)
            printf("%s", about_text);
        else if (choice == 3)
            exit(0);
        else
            printf("Invalid option.\n");
    }
}

int main() {
    menu();
    system("clear");
    game();
}

File: python/textgame.py
#!/usr/bin/env python2
# -*- coding: utf-8 -*-

##########################################
# Python Text Game for Beginners
# Created for a Python introductory class.
# https://github.com/vesche
##########################################

import os
import time


about_text = '''
This is a text adventure game that is played
inside the command prompt of Windows! You
have to stay alive, fight off some zombies,
and try to escape. Good luck...
'''


def main():
    started = True

    while started:
        # clear the screen
        os.system("cls")

        # print menu options and prompt
        print "CMD.EXE ADVENTURE!\n\t1. Play\n\t2. About\n\t3. Quit"
        menu_choice = raw_input('>')

        # act on menu choice
        if menu_choice == '1':
            game()
        elif menu_choice == '2':
            raw_input(about_text)
        elif menu_choice == '3':
            quit()


def game():
    # define intial variables
    playing     = True
    game_map    = ['_'] * 9
    character   = '@'
    pos         = 4

    # get user defined variables
    name = raw_input("\nWhat's your name? ")
    print "Get ready {}!".format(name)

    while playing:
        # clear the screen
        os.system("cls")

        # update character position
        game_map[pos] = character

        # print map and prompt
        print "CMD.EXE ADVENTURE!\n" + "{} - Level 0\n".format(name)
        print ' '.join(game_map[0:3])
        print ' '.join(game_map[3:6])
        print ' '.join(game_map[6:9])
        move = raw_input("\n> ").lower()

        # reset position
        game_map[pos] = '_'

        # act on move
        if move == "north":
            if pos <= 2:
                continue
            pos -= 3

        elif move == "south":
            if pos >= 6:
                continue
            pos += 3

        elif move == "east":
            if pos == 2 or pos == 5 or pos == 8:
                continue
            pos += 1

        elif move == "west":
            if pos == 0 or pos == 3 or pos == 6:
                continue
            pos -= 1

        elif move == "quit":
            playing = False


if __name__ == "__main__":
    main()
